code/obj/item/Bamboo_spike_trap.dm

```diff
@@ -0,0 +1,207 @@
+	var/disarming_time = 2 SECONDS ///how long disarming with a wrench should take
+	var/arming_time = 2 SECONDS ///how long arming should take
+	var/break_down_time = 1 SECONDS ///how long an unarmed trap should take to disassemble with a wrench
+	var/break_down_amount = 3 ///how much bamboo is created when this gets broken down, should be less or equal to the amount needed to create it
+	stamina_damage = 0
+	stamina_cost = 0
+	stamina_crit_chance = 5
+
+	New()
+		..()
+		src.create_reagents(src.reagent_storage)
+
+	examine()
+		. = ..()
+		if (src.armed)
+			. += "<span class='alert'>It looks like it's armed.</span>"
+
+
+	attackby(obj/item/W, mob/user)
+		if(issnippingtool(W))
+			if (src.armed)
+				if (ON_COOLDOWN(user, "disarming_spike_trap", user.combat_click_delay))
+					return
+				playsound(src.loc, 'sound/items/Scissor.ogg', 60)
+				user.visible_message("[user] starts disarming [src]...")
+				var/datum/action/bar/icon/callback/action_bar = new /datum/action/bar/icon/callback(user, src, src.disarming_time,
+				/obj/item/bamboo_spike_trap/proc/disarm,\list(user), W.icon, W.icon_state, "[user] finishes disarming [src]")
+				actions.start(action_bar, user)
+				return
+			else
+				if (ON_COOLDOWN(user, "dismantling_spike_trap", user.combat_click_delay))
+					return
+				playsound(src.loc, 'sound/items/Scissor.ogg', 60)
+				user.visible_message("[user] starts breaking down [src]...")
+				var/datum/action/bar/icon/callback/action_bar = new /datum/action/bar/icon/callback(user, src, src.break_down_time,
+				/obj/item/bamboo_spike_trap/proc/break_down,\list(user), W.icon, W.icon_state, "[user] finishes breaking down [src]")
+				actions.start(action_bar, user)
+				return
+		if(istype(W, /obj/item/reagent_containers/glass/))
+			if(!W.reagents.total_volume)
+				boutput(user, "<span class='alert'>There is nothing in [W] to lace [src] with!</span>")
+				return
+			else
+				var/transferable_amount = min(W:amount_per_transfer_from_this, W.reagents.total_volume, src.reagents.maximum_volume - src.reagents.total_volume)
+				if (transferable_amount <= 0)
+					boutput(user, "<span class='alert'>[src] cannot hold any more chemicals!</span>")
+					return
+				user.visible_message("<span class='notice'>[user] laces [src] with [transferable_amount] units of [W]'s contents.</span>")
+				playsound(src.loc, 'sound/impact_sounds/Liquid_Slosh_1.ogg', 25, 1)
+				W.reagents.trans_to(src, transferable_amount)
+				return
+		..()
+
+	ex_act(severity)
+		//no reuseable explosive chem traps, sorry
+		qdel(src)
+
+	proc/arm(mob/User)
+		if (!src)
+			return
+		var/trap_occupied = 0
+		for(var/obj/item/B in get_turf(src))
+			if (istype(B, /obj/item/bamboo_spike_trap))
+				var/obj/item/bamboo_spike_trap/BM = B
+				if (BM.armed)
+					trap_occupied = 1
+		if (trap_occupied)
+			if (User)
+				boutput(User, "<span class='alert'>A trap is already placed here!</span>")
+			return
```

Comment:
move the message and return into the loop so you stop when the trap is found and also don't need the `trap_occupied` var